### PR TITLE
Make code more reusable in other contexts

### PR DESCRIPTION
**Pull request by qfel**
- Export processStream methods, they are useful when using the code
outside of Istio
- Move verifySentResourcesMultipleTypes to client_test.go

### CODE REVIEW

File: pkg/mcp/internal/size.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Provides a ProtoSize method that can be swapped for proto.Size when using Google protos.
```
Comment (ayj) [inline pkg/mcp/internal/size.go]: Missing the full copyright header (see other files for ref)

**Comment by andraxylia** [comment]
This change just exports the function ProcessStream and moves a function to another file, so it does not have impact on the release.

**Comment by ozevren** [comment]
Let's land this after the merge of release-1.1 into master.

**Comment by qfel** [comment]
Let me know once you want to review it and I'll look into updating this PR.

**Comment by ozevren** [comment]
@qfel can you rebase?
@ayj can you review after the rebase?
Thanks

**Comment by ayj** [comment]
```
pkg/mcp/source/source_test.go:444:6: s.processStream undefined (type *Source has no field or method processStream, but does have ProcessStream)
pkg/mcp/source/source_test.go:476:12: s.processStream undefined (type *Source has no field or method processStream, but does have ProcessStream)
pkg/mcp/source/source_test.go:789:14: s.processStream undefined (type *Source has no field or method processStream, but does have ProcessStream)
```

**Comment by ayj** [comment]
Assuming tests will be fixed before the bot merges,

/lgtm

**Comment by ozevren** [comment]
@qfel there are build failures. Please fix them.

**Comment by qfel** [comment]
Yes, I'm need some other code to go in first (because bumping Istio version) to make sure this PR still works as expected.

**Comment by qfel** [comment]
OK, should be ready now

**Pull request merged**